MedicMeditationApp/Presentation/SignIn/SignInViewModel.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  SignInViewModel.swift
+//  MedicMeditationApp
+//
+//  Created by Ronaldo Andre on 22/05/24.
+//
+
+import Foundation
+import Combine
+import Dispatch
+
+@MainActor
+
+final class SignInViewModel: ObservableObject {
+    
+    let loginRepository : LoginRepository
+    
+    var cancellables = Set<AnyCancellable>()
+    
+    @Published private(set) var loginState = LoginUiState.inicial
+    
+    init(loginRepository: LoginRepository) {
+        self.loginRepository = loginRepository
+    }
+    
+    func startLogin(correo: String, password: String) {
+        
+        guard !correo.isEmpty else {
```

Comment:
Agregale una condicion mas que el teto ingresado sea de una formato de correo valido. Link de referencia: https://stackoverflow.com/questions/25471114/how-to-validate-an-e-mail-address-in-swift